tests/distributed/test_distributed_sampling.py

```diff
@@ -0,0 +1,87 @@
+import dgl
+import unittest
+import os
+from dgl.data import CitationGraphDataset
+from dgl.distributed.sampling import sample_neighbors
+from dgl.distributed import partition_graph, load_partition, GraphPartitionBook
+import sys
+import multiprocessing as mp
+import numpy as np
+import backend as F
+import time
+from utils import get_local_usable_addr
+from pathlib import Path
+
+from dgl.distributed import DistGraphServer, DistGraph
+
+def myexcepthook(exctype, value, traceback):
+    for p in mp.active_children():
+        p.terminate()
+    # raise Exception("11111111111")
+
+
+
+def start_server(rank, tmpdir):
+    import dgl
+    g = DistGraphServer(rank, "rpc_sampling_ip_config.txt", 1, "test_sampling",
+                        tmpdir / 'test_sampling.json')
+    g.start()
+
+def start_client(rank, tmpdir):
+    import dgl
+    g = DistGraph("rpc_sampling_ip_config.txt", "test_sampling")
+    print("Pre sample")
+    print(g.number_of_nodes())
+    results = sample_neighbors(g, [0, 10, 99, 66, 1024, 2008], 3)
+    print("after sample")
+    dgl.distributed.shutdown_servers()
+    dgl.distributed.finalize_client()
+    return results
+
+
+@unittest.skipIf(os.name == 'nt', reason='Do not support windows yet')
+@unittest.skipIf(dgl.backend.backend_name != 'PyTorch', reason='Only support pytorch for now')
```

Comment:
does it not work with mxnet?

Reply:
I heard MXNet 2.0 will break lots of backward compatibility. I think we can support MXNet later

Reply:
but we are not using mxnet 2.0.

Reply:
how difficult is it to make mxnet work?